modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb

```diff
@@ -47,14 +47,17 @@ def submit_form_2122 # rubocop:disable Metrics/MethodLength
             power_of_attorney.save!
           end
 
-          # This job only occurs when a Veteran submits a POA request, they are not required to submit a document.
-          ClaimsApi::PoaUpdater.perform_async(power_of_attorney.id) unless header_request?
           if enable_vbms_access?
-            ClaimsApi::VBMSUpdater.perform_async(power_of_attorney.id,
-                                                 target_veteran.participant_id)
+            ClaimsApi::VBMSUpdater.perform_async(power_of_attorney.id, target_veteran.participant_id)
           end
+
           data = power_of_attorney.form_data
-          ClaimsApi::PoaFormBuilderJob.perform_async(power_of_attorney.id) if data['signatures'].present?
+
+          if data.dig('signatures', 'veteran').present? && data.dig('signatures', 'representative').present?
+            # Autogenerate a 21-22 form from the request body and upload it to VBMS.
+            # If upload is successful, then the PoaUpater job is also called to update the code in BGS.
+            ClaimsApi::PoaFormBuilderJob.perform_async(power_of_attorney.id, poa_code)
```

Comment:
I'm surprised this isn't blowing up because it looks like `poa_code` is being passed as a 2nd arg, but the `perform` method signature is defined to only accept 1 arg.  🤔

Reply:
good catch, removing it in this next commit. I'm also surprised it didn't cause tests to fail